src/screens/Classroom/ClassroomHome.js

```diff
@@ -33,39 +35,68 @@ const styles = StyleSheet.create({
   },
 });
 
-/* const post1 = {
-  author: 'Timmy Turner',
-  text: "I wish that my code doesn't break I really want some pie
-  right now I wonder how people are gopnna see what im typing pls
-  bro just i want to eat i rlly want food bro pls",
-  time: '10:16 PM',
-};
+// eslint-disable-next-line no-unused-vars
+export default function ClassroomHome({ navigation }) {
+  const [initializing, setInitializing] = useState(true);
+  const uid = navigation.getParam('uid', null);
+  const classroomInfo = navigation.getParam('classroomInfo', null);
+  const code = navigation.getParam('code', null);
+  const ref = Firestore().collection('users');
+  const [userState, setUserState] = useState(INITIAL_USER_STATE);
 
-const cl = {
-  classroom: {
-    title: 'CL1',
-    posts: [post1],
-  },
-}; */
+  // const [showLoading, setShowLoading] = useState(false);
+
+  async function getUserData() {
+    try {
+      // If we somehow get to this screen with no uid passed, go back to homescreen
+      if (!uid) navigation.navigate('Home');
+      const doc = await ref.doc(uid).get();
+      const data = doc.data();
+      // Handler for case with nonexistent user entries in Firestore
+      if (!data) {
+        await ref.doc(uid).set({
+          ...userState,
+          createdAt: Firestore.FieldValue.serverTimestamp(),
+          updatedAt: Firestore.FieldValue.serverTimestamp(),
+        });
+      } else {
+        setUserState(data);
+      }
+      if (initializing) setInitializing(false);
+    } catch (e) {
+      setInitializing(false);
+      Alert.alert(
+        e.message,
+      );
+    }
+  }
+
+  useEffect(() => {
+    getUserData();
+  }, [userState]);
+
+  if (initializing) return null;
```

Comment:
What is the purpose of the `initializing` state variable? What happens if you remove it?